qbraid/runtime/azure/provider.py

```diff
@@ -0,0 +1,494 @@
+
+    # pylint: disable=too-many-arguments
+    @staticmethod
+    def create_job_routes(
+        session: str,
+        container: str,
+        qasm: str,
+        connection_string: str,
+        location_name: str,
+        subscription_id: str,
+        resource_group_name: str,
+        workspace_name: str,
+        quantum_access_token: str,
+    ) -> list[str]:
+        """Create the routes for the job."""
+        containerName = container
+        qasm_str = qasm
+
+        blob_service_client = BlobServiceClient.from_connection_string(connection_string)
+
+        blob_client = blob_service_client.get_blob_client(containerName, "inputData")
+        blob_client.upload_blob(
+            qasm_str, content_settings=ContentSettings(content_type="application/qasm")
+        )
+
+        blob_client = blob_service_client.get_blob_client(containerName, "rawOutputData")
+        blob_client.upload_blob("")
+
+        url = (
+            f"https://{location_name}.quantum.azure.com/subscriptions/{subscription_id}"
+            f"/resourceGroups/{resource_group_name}/providers/Microsoft.Quantum/workspaces/"
+            f"{workspace_name}/storage/sasUri?api-version=2022-09-12-preview"
+        )
+
+        quantum_headers = {
+            "Content-type": "application/json",
+            "Authorization": f"Bearer {quantum_access_token}",
+        }
+
+        payload = {"containerName": containerName}
+
+        response = session.post(url, json=payload, headers=quantum_headers)
+        containerSasUri = response.json()["sasUri"]
+
+        payload["blobName"] = "inputData"
+        response = session.post(url, json=payload, headers=quantum_headers)
+        inputSasUri = response.json()["sasUri"]
+
+        payload["blobName"] = "rawOutputData"
+        response = session.post(url, json=payload, headers=quantum_headers)
+        outputSasUri = response.json()["sasUri"]
+
+        return [containerSasUri, inputSasUri, outputSasUri, quantum_headers]
+
+    # pylint: disable=too-many-arguments
+    @staticmethod
+    def create_payload(
+        container_uri: str,
+        input_uri: str,
+        output_uri: str,
+        job_id: str,
+        job_name: str,
+        input_data_format: str,
+        provider: str,
+        backend: str,
+        num_qubits: str,
+        num_shots: int,
+        total_count: int,
+        output_data_format: str,
+    ) -> dict:
+        """Create the payload for the job."""
+        payload = {
+            "containerUri": container_uri,
+            "id": job_id,
+            "inputDataFormat": input_data_format,  # "honeywell.openqasm.v1"
+            "itemType": "Job",
+            "name": job_name,
+            "providerId": provider,
+            "target": backend,  # changeable
+            "metadata": {
+                "qiskit": "True",
+                "name": job_name,
+                "num_qubits": num_qubits,
+                "metadata": "null",
+                "meas_map": "[0]",
```

Comment:
use `json.dumps` to convert python types to string format. Don't need to encode as strings yourself